python/lsst/pipe/tasks/makeCoaddTempExp.py

```diff
@@ -438,12 +447,18 @@ def getCalibratedExposure(self, dataRef, bgSubtracted):
                                   calexp's background background model and add it to the calexp.
         @return calibrated exposure
 
+        @raises MissingExposureError If data for the exposure is not available.
```

Comment:
Good catch on the name. I bounced back and forth between those a few times.

In this case, this "catch and raise" will make more sense with DM-16537. The idea is to better follow the logic you described to me on slack: sometimes exposures don't exist in a patch, and you want to ignore that. However, other errors should not be masked, for example `ImportError` for meas_mosaic, the jointcal calibrations being missing (jointcal persists all exposures it processes in each visit, even if the Exposure isn't in the tract), or a legitimate math error. This makes the distinction between those cases explicit.

I've also wrapped the meas_mosaic line in a similar try:except, because my understanding is that mosaic isn't guaranteed to persist every Exposure in a visit.